source/MaterialXTest/Render.cpp

```diff
@@ -1212,20 +1211,20 @@ struct ImageHandlerTestOptions
 void testImageHandler(ImageHandlerTestOptions& options)
 {
     mx::FilePath imagePath = mx::FilePath::getCurrentPath() / mx::FilePath("resources/Images/");
-    mx::StringVec files;
+    mx::FilePathVec files;
 
     unsigned int loadFailed = 0;
-    for (auto extension : options.testExtensions)
+    for (const std::string& extension : options.testExtensions)
     {
         if (options.skipExtensions.end() != std::find(options.skipExtensions.begin(), options.skipExtensions.end(), extension))
         {
             continue;
         }
-        mx::getFilesInDirectory(imagePath, files, extension);
-        for (const std::string& file : files)
+        files = imagePath.getFilesInDirectory(extension);
+        for (const mx::FilePath& file : files)
         {
-            const mx::FilePath filePath = imagePath / mx::FilePath(file);
-            const std::string fileName = filePath;
+            const mx::FilePath filePath = imagePath / file;
+            const std::string& fileName = filePath;
```

Comment:
Can a FilePath be converted to a string _reference_? Will it be a reference to a temporary local copy of this string?

But maybe fileName could be removed as it's only used for logging below. Could use filePath.asString() for the logging instead?